From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Fri, 4 Dec 2015 08:57:30 +0100
Subject: Add multiarch triplet as optional search path

CMAKE_LIBRARY_ARCHITECTURE needs to be defined by the project using
pluginlib. Useful for libraries using pluginlib.
---
 include/pluginlib/class_loader_imp.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/pluginlib/class_loader_imp.h b/include/pluginlib/class_loader_imp.h
index 38f1db5..ac19be9 100644
--- a/include/pluginlib/class_loader_imp.h
+++ b/include/pluginlib/class_loader_imp.h
@@ -248,6 +248,9 @@ namespace pluginlib
     std::vector<std::string> all_paths_without_extension = getCatkinLibraryPaths();
     all_paths_without_extension.push_back(getROSBuildLibraryPath(exporting_package_name));
     all_paths_without_extension.push_back("/usr/lib/" + exporting_package_name);
+#ifdef CMAKE_LIBRARY_ARCHITECTURE
+    all_paths_without_extension.push_back("/usr/lib/" CMAKE_LIBRARY_ARCHITECTURE);
+#endif
     bool debug_library_suffix = (class_loader::systemLibrarySuffix().compare(0, 1, "d") == 0);
     std::string non_debug_suffix;
     if(debug_library_suffix) {
